client/src/components/timeline/revision.container.js

```diff
@@ -1,9 +1,23 @@
 import { connect } from 'react-redux';
-import * as WikiSelectors from 'app/selectors/wiki';
+import { toggleDiff } from 'app/actions/diff';
+import { makeGetDiff } from 'app/selectors/diff';
+import { makeGetRevisionUrl, makeGetTimestamp } from 'app/selectors/revisions';
+import makeGetSide from 'app/selectors/side';
 import Revision from './revision';
 
+const getSide = makeGetSide();
+const getRevisionUrl = makeGetRevisionUrl();
+const getTimestamp = makeGetTimestamp();
+const getDiff = makeGetDiff();
+
 export default connect(
-	state => ( {
-		wiki: WikiSelectors.getWiki( state )
-	} )
+	( state, props ) => ( {
+		side: getSide( state, props ),
+		url: getRevisionUrl( state, props ),
+		timestamp: getTimestamp( state, props ),
+		diff: getDiff( state, props )
+	} ),
+	dispatch => ( {
+		toggleDiff: ( diff, suppressed ) => dispatch( toggleDiff( diff, suppressed ) )
```

Comment:
Sorcery!! can you explain me how a dispatch inside a dispatch works?

Reply:
so
```
dispatch => ( {

} )
```
is just a function that takes a single parameter, a function, named `dispatch` and returns an object.

which means:
```
( diff, suppressed ) => dispatch( toggleDiff( diff, suppressed ) )
```
is a function that takes in two parameters, but uses the `dispatch` that was passed in from the higher scope.

basically, scope is magic. :P